src/FTPClient.jl

```diff
@@ -16,6 +16,7 @@ type FTPClientError <: Exception
        lib_curl_error::UInt32
 end
 
+@enum FTP_MODE ascii_mode binary_mode
```

Comment:
A user doesn't need to know about `TRANSFER`, unless they are making their own functions that are using those enum types. But I do agree that `module` is overkill for this. I'll keep it as `@enum` and `ascii_mode` and `binary_mode` for this PR.